pkg/util/merger.go

```diff
@@ -43,3 +43,40 @@ func MergeNSampleSets(sampleSets ...[]model.SamplePair) []model.SamplePair {
 	right := MergeNSampleSets(sampleSets[n:]...)
 	return MergeSampleSets(left, right)
 }
+
+func MergeSortedStringLists(lnss [][]string) []string {
```

Comment:
I was hoping we already had an implementation of this, but I can't find it.

We might want to look at `LabelValuesForLabelName` and use the same method there?

Reply:
We have something similar in here, but for `model.LabelNames`. Direct typecast was not possible, so decided to have another one for `[]string`.
https://github.com/cortexproject/cortex/blob/f6627227636f447b29cd46e0ec3f432ed0a1a0d3/pkg/ingester/index/index.go#L299

`LabelValuesForLabelName` uses map to filter out, and should be also apt for label names as they are less in number. I think I will remove this function and use maps everywhere instead.